API.md

```diff
@@ -0,0 +1,196 @@
+
+## **Initiate Dependency Analysis**
+
+Submits a dependency stack analysis request
+
+- **URL**
+  `https://<3scale-api-gateway-url>/api/v1/stack-analyses`
+
+- **Method**
+  `POST`
+
+- **Response Format**
+  `jSON`
+
+- **Path Parameters**
+  `None`
+
+- **Query Parameters**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | user_key | required | 3-scale API management authorization user key. | 421249d63 |
+
+- **Form Data**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | manifest[] | required | Ecosystem specific Manifest file(s) available in the application. One can provide multiple manifest files to invoke the method. | @pom.xml |
+  | filePath[] | required | Corresponding manifest file’s directory information. Care has to be taken to map the manifest filenames and file paths correctly. | /home/JohnDoe |
+
+- **Headers**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | source | required | Source of the request. | "vscode" |
+  | ecosystem | required | Ecosystem | maven/pypi/npm |
+
+- **Sample Request with cURL**
+
+  ``` 
+  $ curl -k -F "manifest[]=@./pom.xml" -F "filePath[]=/home/JohnDoe" https://https://api-244552094075730.prod.gw.apicast.io:443/api/v1/stack-analyses/?user_key=421249d63
+  ```
+
+- **Sample Responses**
+
+  **_Success_**
+
+    ```
+        {
+        "id": "421249d9e1e5464cbf3e77dde4941463",
+        "status": "success",
+        "submitted_at": "2017-10-20 05:09:01.165068"
+        }
+    ```
+
+  **_Failure_**
+    ```
+        400:
+            {
+                "error": "Bad request"
+            }
+    ```
+
+    ```
+        401:
+            {
+                "error": "Authentication failed - could not decode JWT token"
+            }
+    ```
+
+## **Fetch Dependency Analysis**
+
+Fetches dependency stack analysis response using the request identifier returned by initiate dependency analysis.
+
+- **URL**
+  `https://<3scale-api-gateway-url>/api/v1/stack-analyses/{request ID}`
+
+- **Method**
+  `GET`
+
+- **Response Format**
+  `jSON`
+
+- **Path Parameters**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | {request ID} | required | Request identifier returned by /stack-analyses POST. | 421249d9e1e5464cbf3e77dde4941463 |
+
+- **Query Parameters**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | user_key | required | 3-scale API management authorization user key. | 421249d63 |
+
+- **Sample Request with cURL**
+
+  ``` 
+  $ curl -k https://https://api-244552094075730.prod.gw.apicast.io:443/api/v1/stack-analyses/421249d9e1e5464cbf3e77dde4941463 
+  ```
+
+- **Sample Responses**
+
+  **_Success_**
+
+    Sample success reponse is [here](https://gist.github.com/sivaavkd/1b27d01ccd17af839c0ba634ce9de628)
+
+  **_In Progress_**
+    ```
+        200:
```

Comment:
This will be 202. In case of 200, user gets to see the report

Reply:
Thanks Mitesh.  Updated it to 202.

Reply:
As you suggested, also added 202 for component analysis.